batch_requests/views.py

```diff
@@ -89,6 +91,14 @@ def construct_wsgi_from_data(data):
 
         body = data.get("body", "")
         headers = data.get("headers", {})
+
+        '''
+            Add token manually to sub-requests as there's a problem with automatic sub-requests'
+            session auth 
+        '''
+        token, created = Token.objects.get_or_create(user=request.user)
```

Comment:
When not using a variable like `created`, it's generally best to assign it to `_`. Linters usually don't flag that as an unused variable. Some old django projects use `_` for a translation package, though, so check the imports.

`Token.objects.get_or_create(user=request.user)` will generate a db call for each sub-request, but you're getting the same token every time. Generate the token string outside the builder function, then assign it to the header inside the builder function.

Reply:
Solved https://github.com/GoLorry/django-batch-requests/pull/1/commits/cdd8210e85f80eb2ddb732aaeac90d97e3567d03